libnetdata/queue/queue.h

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#ifndef NETDATA_QUEUE_H
+#define NETDATA_QUEUE_H 1
+
+#include "stdlib.h"
+#include "pthread.h"
+
+typedef struct nodes *node;
+typedef struct queues *queue;
+
+extern queue initqueue(int max);
+extern void freequeue(queue q);
+extern int enqueue(queue, void*);	
+extern void* dequeue(queue);				
```

Comment:
I suggest the following API:

```
queue_t queue_new(int size);
void queue_free(queue_t q);
int queue_push(queue_t q, void *data);
void *queue_pop(queue_t q);
```

ie:

- add the `_t` suffix on typedefs so that we can differentiate types from variables,
- namespace symbols under `queue_*`
- drop `extern`, it's used in other parts of the codebase but it's an anachronism because all function decls are extern by default in C.